File: src/main/java/ar/edu/unq/spring/persistence/ItemDAO.java
package ar.edu.unq.spring.persistence;

import ar.edu.unq.spring.modelo.Item;

import java.util.Collection;
import java.util.Optional;
import java.util.Set;


public interface ItemDAO {
    Item save(Item item);
    Optional<Item> findById(Long id);
    Collection<Item> findAll();
    void deleteAll();
    void delete(Item item);
    Item findTopByOrderByPesoDesc();
    Set<Item> getMasPesados(int peso);
    Set<Item> getItemsDePersonajesDebiles(int vida);
}

File: src/main/kotlin/ar/edu/unq/spring/persistence/ItemDAO.kt
package ar.edu.unq.spring.persistence

import ar.edu.unq.spring.modelo.Item
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository

interface ItemDAO : CrudRepository<Item, Long> {

    fun findTopByOrderByPesoDesc(): Item

    @Query(
        "from Item i "
                + "where i.owner.vida < ?1 "
                + "order by i.peso asc"
    )
    fun getItemsDePersonajesDebiles(unaVida: Int): Collection<Item>


    @Query(
        "FROM Item i where i.peso  > ?1 order by i.peso asc"
    )
    fun getMasPesados(unValorDado: Int): Collection<Item>

}
